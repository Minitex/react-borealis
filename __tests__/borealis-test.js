import React from 'react';
import renderer from 'react-test-renderer';
import Borealis from '../src/react-borealis';

jest.mock('react-openseadragon', () => 'OpenSeadragonViewer');

describe('Borealis', () => {
  it('should render correctly', () => {
    const config = {
      kaltura_audio_playlist: {
        targetId: 'kaltura_player_1489684320',
        wid: '_1369852',
        uiconf_id: 38719361,
        flashvars: {
          streamerType: 'auto',
          'playlistAPI.kpl0Id': '0_0s5bpfh7',
        },
        transcript: {
          texts: ['this is a kaltura video transcript'],
          label: 'Video',
        },
        height: '395px',
        width: '560px',
        thumbnail: 'http://reflections.mndigital.org/utils/getthumbnail/collection/jhs/id/885',
      },
      kaltura_video: {
        targetId: 'kaltura_player_1489614720',
        wid: '_1369852',
        uiconf_id: 38683631,
        transcript: {
          texts: ['this is a kaltura video transcript'],
          label: 'Video',
        },
        entry_id: '0_t96feg4f',
        height: '315px',
        width: '560px',
        thumbnail: 'http://reflections.mndigital.org/utils/getthumbnail/collection/p16022coll38/id/0',
      },
      kaltura_audio: {
        targetId: 'kaltura_player_1489599711',
        wid: '_1369852',
        uiconf_id: 38708801,
        entry_id: '0_b64f8k0c',
        transcript: {
          texts: ['this is a kaltura transcript'],
          label: 'Audio',
        },
        height: '70px',
        width: '460px',
        thumbnail: 'http://reflections.mndigital.org/utils/getthumbnail/collection/jhs/id/885',
      },
      image: {
        thumbnail: 'https://stacks.stanford.edu/image/iiif/hg676jb4964%252F0380_796-44/full/340,/0/default.jpg',
        type: 'image',
        label: 'Image',
        include_controls: true,
        sequenceMode: true,
        showReferenceStrip: true,
        defaultZoomLevel: 0,
        tileSources: [
          'https://stacks.stanford.edu/image/iiif/hg676jb4964%2F0380_796-44/info.json',
          'https://ids.lib.harvard.edu/ids/iiif/25286610/info.json',
        ],
        transcript: {
          texts: ['First Image Item Transcript', 'Second Image Item  Transcript'],
          label: 'Image',
        },
        tocs: [
          'A Statue',
        ],
      },
      pdf: {
        height: 800,
        src: 'https://s3.amazonaws.com/mdl-assets/stpaul.pdf',
        thumbnail: 'http://reflections.mndigital.org/utils/getthumbnail/collection/p16022coll35/id/0',
        transcript: {
          texts: ['PDF Transcript Here'],
          label: 'PDF',
        },
      },
      ppt: {
        transcript: {
          texts: [],
          label: 'PowerPoint',
        },
        text: '(Download)',
        thumbnail: 'http://reflections.mndigital.org/utils/getthumbnail/collection/p16022coll17/id/827',
        src: 'http://reflections.mndigital.org/utils/getdownloaditem/collection/p16022coll17/id/827/filename/828.pptx',
      },
      audio: {
        src: 'http://reflections.mndigital.org/utils/getstream/collection/mhs/id/1919',
        thumbnail: 'http://reflections.mndigital.org/utils/getthumbnail/collection/jhs/id/885',
        transcript: {
          texts: ['Audio Transcript Here'],
          label: 'Audio',
        },
      },
      video: {
        thumbnail: 'http://reflections.mndigital.org/utils/getthumbnail/collection/p16022coll38/id/0',
        transcript: {
          texts: [],
          label: 'Video',
        },
        height: 500,
        width: 500,
        src: 'http://reflections.mndigital.org/utils/getstream/collection/stc/id/8470',
      },
    };

    const component = renderer.create(<Borealis config={config} basename="/" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
