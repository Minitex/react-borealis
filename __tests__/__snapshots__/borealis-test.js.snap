exports[`Borealis should render correctly 1`] = `
<div>
  <div>
    <div
      className="borealis-tray">
      <div>
        <a
          className="active borealis-thumbnail"
          href="#/image/0"
          onClick={[Function]}
          style={Object {}}>
          <img
            alt="Click to load the image/0"
            src="https://stacks.stanford.edu/image/iiif/hg676jb4964%252F0380_796-44/full/340,/0/default.jpg" />
        </a>
      </div>
      <div>
        <a
          className="borealis-thumbnail"
          href="#/kaltura_audio"
          onClick={[Function]}
          style={undefined}>
          <img
            alt="Click to load the kaltura_audio"
            src="http://reflections.mndigital.org/utils/getthumbnail/collection/jhs/id/885" />
        </a>
      </div>
      <div>
        <a
          className="borealis-thumbnail"
          href="#/kaltura_audio_playlist"
          onClick={[Function]}
          style={undefined}>
          <img
            alt="Click to load the kaltura_audio_playlist"
            src="http://reflections.mndigital.org/utils/getthumbnail/collection/jhs/id/885" />
        </a>
      </div>
      <div>
        <a
          className="borealis-thumbnail"
          href="#/kaltura_video"
          onClick={[Function]}
          style={undefined}>
          <img
            alt="Click to load the kaltura_video"
            src="http://reflections.mndigital.org/utils/getthumbnail/collection/p16022coll38/id/0" />
        </a>
      </div>
      <div>
        <a
          className="borealis-thumbnail"
          href="#/audio"
          onClick={[Function]}
          style={undefined}>
          <img
            alt="Click to load the audio"
            src="http://reflections.mndigital.org/utils/getthumbnail/collection/jhs/id/885" />
        </a>
      </div>
      <div>
        <a
          className="borealis-thumbnail"
          href="#/video"
          onClick={[Function]}
          style={undefined}>
          <img
            alt="Click to load the video"
            src="http://reflections.mndigital.org/utils/getthumbnail/collection/p16022coll38/id/0" />
        </a>
      </div>
      <div>
        <a
          className="borealis-thumbnail"
          href="#/pdf/0"
          onClick={[Function]}
          style={undefined}>
          <img
            alt="Click to load the pdf/0"
            src="http://reflections.mndigital.org/utils/getthumbnail/collection/p16022coll35/id/0" />
        </a>
      </div>
      <div>
        <a
          className="borealis-thumbnail"
          href="#/ppt"
          onClick={[Function]}
          style={undefined}>
          <img
            alt="Click to load the ppt"
            src="http://reflections.mndigital.org/utils/getthumbnail/collection/p16022coll17/id/827" />
        </a>
      </div>
    </div>
    <div
      className="row image-nav">
      <ul
        className="nav nav-pills">
        <li>
          <a
            className="active transcript-nav-link"
            href="#/image/0"
            onClick={[Function]}
            style={Object {}}>
            Image
          </a>
        </li>
        <li>
          <a
            className="transcript-nav-link"
            href="#/image/0/transcript"
            onClick={[Function]}
            style={undefined}>
            Transcript
          </a>
        </li>
      </ul>
    </div>
    <OpenSeadragonViewer
      WrappedComponent="OpenSeadragonViewer"
      basename="/"
      config={
        Object {
          "defaultZoomLevel": 0,
          "include_controls": true,
          "label": "Image",
          "sequenceMode": true,
          "showReferenceStrip": true,
          "thumbnail": "https://stacks.stanford.edu/image/iiif/hg676jb4964%252F0380_796-44/full/340,/0/default.jpg",
          "tileSources": Array [
            "https://stacks.stanford.edu/image/iiif/hg676jb4964%2F0380_796-44/info.json",
            "https://ids.lib.harvard.edu/ids/iiif/25286610/info.json",
          ],
          "tocs": Array [
            "A Statue",
          ],
          "transcript": Object {
            "label": "Image",
            "texts": Array [
              "First Image Item Transcript",
              "Second Image Item  Transcript",
            ],
          },
          "type": "image",
        }
      }
      defaultZoomLevel={0}
      getThumbnailPath={[Function]}
      history={
        Object {
          "action": "REPLACE",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 3,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/image/0",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
      include_controls={true}
      label="Image"
      location={
        Object {
          "hash": "",
          "pathname": "/image/0",
          "search": "",
          "state": undefined,
        }
      }
      match={
        Object {
          "isExact": true,
          "params": Object {
            "id": "0",
          },
          "path": "/image/:id",
          "url": "/image/0",
        }
      }
      sequenceMode={true}
      showReferenceStrip={true}
      staticContext={undefined}
      thumbnail="https://stacks.stanford.edu/image/iiif/hg676jb4964%252F0380_796-44/full/340,/0/default.jpg"
      tileSources={
        Array [
          "https://stacks.stanford.edu/image/iiif/hg676jb4964%2F0380_796-44/info.json",
          "https://ids.lib.harvard.edu/ids/iiif/25286610/info.json",
        ]
      }
      tocs={
        Array [
          "A Statue",
        ]
      }
      transcript={
        Object {
          "label": "Image",
          "texts": Array [
            "First Image Item Transcript",
            "Second Image Item  Transcript",
          ],
        }
      }
      type="image" />
  </div>
</div>
`;
